File: 49-group-anagrams/group-anagrams.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
    Map<String,ArrayList<String>> map = new HashMap<>();

    for(String word : strs )
{
      char[] text= word.toCharArray();
      Arrays.sort(text);
      String te = new String(text);
      if(!map.containsKey(te)){
        map.put(te,new ArrayList<>());
      }
    map.get(te).add(word);
}
 
        return new ArrayList<>(map.values());
        
    }
}

File: 49-group-anagrams/group-anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        res=defaultdict(list)
        for s in strs:
            a=[0]*26
            for c in s:
                a[ord(c)-ord('a')]+=1
            key =tuple(a)
            res[key].append(s)
        return list(res.values())
        